zerver/lib/digest.py

```diff
@@ -375,13 +374,9 @@ def bulk_handle_digest_email(user_ids: List[int], cutoff: float) -> None:
         .order_by("id")
         .select_related("realm")
     )
-    context_map = bulk_get_digest_context(users, cutoff)
-
     digest_users = []
 
-    for user in users:
-        context = context_map[user.id]
-
+    for user, context in bulk_get_digest_context(users, cutoff):
```

Comment:
Just thinking out loud: If something fails, we don't want to be left in a state where emails get double-sent or something. But this operation runs in a transaction, and `send_future_email` puts things in the database, so I think all the side effects will roll back in the event of a failure?

Reply:
Yup.  None of that was changed in this cleanup.